lib/commands/screenshots.js

```
@@ -6,21 +6,26 @@ import log from '../logger';
 let commands = {};
 
 commands.getScreenshot = async function () {
+  const getScreenshotFromWDA = async () => {
+    const data = await this.proxyCommand('/screenshot', 'GET');
+    if (!_.isString(data)) {
+      throw new Error(`Unable to take screenshot. WDA returned '${JSON.stringify(data)}'`);
+    }
+    return data;
+  };
   try {
+    return await getScreenshotFromWDA();
+  } catch (err) {
     if (!this.isRealDevice() && this.xcodeVersion.versionFloat >= 8.1) {
+      log.info(`Falling back to 'simctl io screenshot' API`);
       return await getScreenshot(this.opts.udid);
     }
-  } catch (err) {
-    log.warn(`Cannot make a screenshot using simctl because of "${err.message}". Falling back to WDA API`);
+    let result;
```

I want this behaviour. WDA fails, simctl fails - there is no other way to get the screenshot, so we fail fast not to spend time retrying it.